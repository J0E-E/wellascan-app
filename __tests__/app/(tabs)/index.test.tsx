import React from 'react'
import { render } from '@testing-library/react-native'
import HomeScreen from '@/app/(tabs)'

// Mock Bottom Tab Height
jest.mock('@react-navigation/bottom-tabs', () => ({
	useBottomTabBarHeight: () => 50,
}))

// ParralaxScrollView is not compatable with snapshots. Too Complex.
jest.mock('@/components/ParallaxScrollView', () => {
	const { View } = require('react-native')
	return ({ children }: { children: React.ReactNode }) => <View>{children}</View>
})

// Mock expo-image
jest.mock('expo-image', () => {
	const { View } = require('react-native')
	return {
		Image: View,
	}
})

describe('HomeScreen', () => {
	it('renders and matches snapshot (safe subset)', () => {
		const { toJSON } = render(<HomeScreen />)

		// SAFER than snapshotting container or screen.toJSON() when errors occur
		expect(toJSON()).toMatchSnapshot()
	})
})
