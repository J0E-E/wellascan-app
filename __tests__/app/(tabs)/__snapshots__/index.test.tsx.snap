// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders and matches snapshot (safe subset) 1`] = `
<View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "alignItems": "center",
          "flexDirection": "row",
          "gap": 8,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          {
            "fontSize": 32,
            "fontWeight": "bold",
            "lineHeight": 32,
          },
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Welcome to the Wella App!
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "gap": 8,
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          undefined,
          undefined,
        ]
      }
    >
      Step 1: Create a List
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Go to the Lists tab and create at least one list. This is where your scanned products will go.
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "gap": 8,
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          undefined,
          undefined,
        ]
      }
    >
      Step 2: Select Your List
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Open the list and tap "Scan Products to List" â€” or go to the Barcode tab and pick a list to start scanning into.
    </Text>
  </View>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "gap": 8,
          "marginBottom": 8,
        },
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          undefined,
          undefined,
          undefined,
          {
            "fontSize": 20,
            "fontWeight": "bold",
          },
          undefined,
          undefined,
        ]
      }
    >
      Step 3: Start Scanning!
    </Text>
    <Text
      style={
        [
          {
            "color": "#11181C",
          },
          {
            "fontSize": 16,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ]
      }
    >
      Scan Wella products as you use them to keep your reorder list up to date.
    </Text>
  </View>
</View>
`;
